commit 2e51a5dc8768591e1e700107f4e19cd412aa7874
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2019-06-11T14:20:07-07:00 (4 months ago)
    
    TRITON-1719 sdc-migrate abort fails if there's no destination CN
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>

diff --git a/bin/sdc-migrate b/bin/sdc-migrate
index 81eb2a6..6e7f923 100755
--- a/bin/sdc-migrate
+++ b/bin/sdc-migrate
@@ -527,7 +527,17 @@ EOF
 
     local url="/vms/${vm_uuid}?action=migrate&migration_action=abort"
 
-    job_uuid=$(sdc-vmapi "$url" -X POST | $JSON -H job_uuid message)
+    local result=$(sdc-vmapi "$url" -X POST)
+
+    # A 200 status code means the migration was aborted, but it didn't need
+    # a job to delete the target instance (as their was no target instance).
+    local status_code=$(echo "$result" | grep '^HTTP' | head -1 | awk '{print $2}')
+    if [[ $status_code == "200" ]]; then
+        echo "Migration was successfully aborted"
+        return 0
+    fi
+
+    local job_uuid=$($JSON -H job_uuid message <<< "$result")
 
     check_job_uuid "${job_uuid}"
     migration_watch "${vm_uuid}"
